x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinder.java

```diff
@@ -529,16 +536,25 @@ static boolean canCreateFromSample(List<String> explanation, String sample, int
                         --fieldsInThisRow;
                     }
 
-                    if (fieldsInLastRow != fieldsInFirstRow) {
-                        explanation.add("Not " + formatName + " because row [" + (numberOfRows - 1) +
-                            "] has a different number of fields to the first row: [" + fieldsInFirstRow + "] and [" +
-                            fieldsInLastRow + "]");
-                        return false;
+                    if (fieldsInThisRow != fieldsInFirstRow) {
```

Comment:
So in your solution the size of the first row determines what is "normal", right?
If the first row has 5 fields and the rest of the rows have 6 fields each, the file will be rejected even though there is only one "odd" row?
I guess that's ok (it makes the code simpler) but wanted to make sure it is a conscious decision.

Reply:
Delimited parsing needs some sort of anchor to determine where things stand. The first row is usually a nice one as many files contain their header info. 

I am not sure if we should anchor towards the first row or not, but that is how it was before my change :).

Reply:
I think it's OK for this iteration, but remember before this change every single row had to have the same number of fields, so the fact that the code achieved this by anchoring to the number of fields in the first row was an implementation detail.  After this change it becomes the externally visible policy.  But since it would complicate things a lot to change that and the new way of doing things is more tolerant than the old way, let's leave this as-is and see how many people cannot analyze a file because the first row has a different number of fields.